src/argilla/utils/modeling.py

```diff
@@ -0,0 +1,225 @@
+
+        Args:
+            df (pd.DataFrame): The text descriptives dataframe.
+
+        Returns:
+            List: A list of metadata properties.
+        """
+        ### TO DO: Handle nans
+        properties = []
+        for col, dtype in df.dtypes.items():
+            name = f"{self.clean_column_name(col)}"
+            title = name.replace("_", " ").title()
+            if dtype == "object":
+                prop = TermsMetadataProperty(name=name, title=title, visible_for_annotators=self.visible_for_annotators)
+            elif dtype == "int64":
+                prop = IntegerMetadataProperty(
+                    name=name, title=title, visible_for_annotators=self.visible_for_annotators
+                )
+            elif dtype == "float64":
+                prop = FloatMetadataProperty(name=name, title=title, visible_for_annotators=self.visible_for_annotators)
+            elif dtype == "bool":
+                prop = TermsMetadataProperty(name=name, title=title, visible_for_annotators=self.visible_for_annotators)
+            else:
+                print(f"Unhandled data type for column {col}: {dtype}")
+                continue
+            properties.append(prop)
+        return properties
+
+    def add_text_descriptives_to_metadata(
+        self, records: List[FeedbackRecord], df: pd.DataFrame
+    ) -> List[FeedbackRecord]:
+        """
+        Add the text descriptives metrics extracted previously as metadata
+        to a list of FeedbackDataset records.
+
+        Args:
+            records (List[FeedbackRecord]): A list of FeedbackDataset records.
+            df (pd.DataFrame): The text descriptives dataframe.
+
+        Returns:
+            List[FeedbackRecord]: A list of FeedbackDataset records with updated metadata.
+        """
+        # Loop through the records and add the metrics to the metadata
+        modified_records = []
+        for record, metrics in tqdm(zip(records, df.to_dict("records")), total=len(records)):
+            record.metadata.update(metrics)
+            modified_records.append(record)
+        return modified_records
+
+    # def update_records(self, records: List[FeedbackRecord]) -> List[FeedbackRecord]:
+    #     """Updates the metadata of a list of feedback records with text descriptives metrics
+    #     and returns a list of updated feedback records.
+    #     """
+    #     # Create a dataset from the records
+
+    #     # Extract the metrics
+    #     extracted_metrics = self.extract_metrics(records)
+    #     # Create metadata properties based on dataframe columns and data types
+    #     metadata_properties = self.create_metadata_properties(extracted_metrics, prefix="text_descriptives")
+    #     # Add the metrics to the metadata of the records
+    #     modified_records = self.add_text_descriptives_to_metadata(records, extracted_metrics)
+    #     return modified_records
+
+    def update_dataset(self, dataset: FeedbackDataset) -> FeedbackDataset:
```

Comment:
This function should just add/update metadata property config, 
https://docs.argilla.io/en/latest/practical_guides/create_dataset.html#define-metadata-properties
https://docs.argilla.io/en/latest/practical_guides/update_dataset.html#add-or-delete-metadata-properties

Reply:
This method now uses `.add_metadata_property()` to add metadata properties to the existing dataset.